docs/SPEC.md

```diff
@@ -729,29 +729,45 @@ It is not possible to access an object's property using an arbitrary expression.
     DotExpression           = "." identifer
     MemberBracketExpression = "[" string_lit "]" .
 
+#### Conditional Expressions
+
+Conditional expressions evaluate a boolean-valued condition and if the result it _true_,
+the expression following the `then` keyword is evaluated and returned.
+Otherwise the condition evaluates to _null_ or _false_,
+and the expression following the `else` keyword is evaluated and returned.
+
+    ConditionalExpression   = "if" Expression "then" Expression "else" Expression .
+
+Example:
+
+    color = if code == 0 then "green" else if code == 1 then "yellow" else "red"
+
 #### Operators
 
 Operators combine operands into expressions.
 The precedence of the operators is given in the table below. Operators with a lower number have higher precedence.
 
-|Precedence| Operator |        Description        |
-|----------|----------|---------------------------|
-|     1    |  `a()`   |       Function call       |
-|          |  `a[]`   |  Member or index access   |
-|          |   `.`    |       Member access       |
-|     2    | `*` `/`  |Multiplication and division|
-|     3    | `+` `-`  | Addition and subtraction  |
-|     4    |`==` `!=` |   Comparison operators    |
-|          | `<` `<=` |                           |
-|          | `>` `>=` |                           |
-|          |`=~` `!~` |                           |
-|     5    |  `not`   | Unary logical expression  |
-|     6    |  `and`   |        Logical AND        |
-|     7    |  `or`    |        Logical OR         |
+|Precedence| Operator       |        Description        |
+|----------|----------------|---------------------------|
+|     1    |  `a()`         |       Function call       |
+|          |  `a[]`         |  Member or index access   |
+|          |   `.`          |       Member access       |
+|     2    | `*` `/`        |Multiplication and division|
+|     3    | `+` `-`        | Addition and subtraction  |
+|     4    |`==` `!=`       |   Comparison operators    |
+|          | `<` `<=`       |                           |
+|          | `>` `>=`       |                           |
+|          |`=~` `!~`       |                           |
+|     5    |  `not`         | Unary logical expression  |
+|     6    |  `and`         |        Logical AND        |
+|     7    |  `or`          |        Logical OR         |
+|     8    | `if/then/else` |        Conditional        |
```

Comment:
Can you explain the reason for this decision with an example? It doesn't have to be in the document, I'm just curious. I think this is correct, but I'd like to see what the prior art is before I just go with my feeling on the matter.

Reply:
I was basing the keywords on SML syntax.  I chose the precedence intuitively, but you're right to ask about prior art.  The definition of SML is here:
http://sml-family.org/sml97-defn.pdf
On page 75 there is a discussion of grammar and precedence.  In their notation, forms are listed in decreasing precedence.  They give infix operator expressions higher precedence than others, including `if/then/else`.  Only `while`, `case` and `fn` have lower precedence. (This last one is interesting because it seems different from the precedence of function expressions in Flux)

Another example would be C, in which the `?:` conditional expression has lower precedence than everything except the compound assignments (`+=`, etc) and the `,` operator.